scripts/run_docker

```diff
@@ -48,13 +48,28 @@ if [ "$OSTYPE" == "msys" ]; then
   CONTAINER_RUNTIME="winpty docker"
 fi
 
-RAND_NAME=$(env LC_ALL=C tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 16 | head -n 1)
-if [ -z "$NETWORK_NAME" ]; then
-  echo "No Docker network specified."
-  $CONTAINER_RUNTIME run --rm -ti --name "aries-cloudagent-runner_${RAND_NAME}" \
-    $ARGS aries-cloudagent-run "$@"
+ACAPY_CONTAINER_NAME="${CONTAINER_NAME}"
+if [ -n "${CONTAINER_NAME}" ]; then
+  ARGS="${ARGS} --name ${CONTAINER_NAME}"
+else
+  RAND_NAME=$(env LC_ALL=C tr -dc 'a-zA-Z0-9' </dev/urandom | fold -w 16 | head -n 1)
+  ACAPY_CONTAINER_NAME="aries-cloudagent-runner_${RAND_NAME}"
+  ARGS="${ARGS} --name ${ACAPY_CONTAINER_NAME}"
+fi
+echo ""
+echo "Container name: ${ACAPY_CONTAINER_NAME}"
+
+ACAPY_NETWORK_NAME="${NETWORK_NAME}"
+# Create new Docker network if it does not exist.
+if [ -n "${NETWORK_NAME}" ] && [ -z "$(docker network ls --filter name=^"${NETWORK_NAME}"$ --format="{{ .Name }}")" ]; then
```

Comment:
I believe this should use `$CONTAINER_RUNTIME` instead of `docker`. It might be cleaner to have one outer block for `[ -n "${NETWORK_NAME}" ]` and check for the existence of the network inside.

Reply:
Thank you for pointing this. Fixed.